### PR TITLE
Restructure v2.6 Security section 

### PR DESCRIPTION
**Pull request by macedogm**
This is an initial work preparing for the updates to CIS hardening guides for v2.6 - https://github.com/rancher/rancher/issues/35735.

1. Remove v2.5 related hardening guides from v2.6 and mentioned that v2.6 guides are being updated.
2. Added  aliases for removed pages.
3. Update Kubernetes security best practices.
4. Fix some small typos.

### CODE REVIEW

**Comment by vincent99** [comment]
Removed or renamed files == broken links in search engines indefinitely.  Setup redirects in frontmatter to preserve all existing paths and go to wherever is appropriate.

**Comment by macedogm** [comment]
@vincent99 thanks for catching this. Is this - https://github.com/rancher/docs/blob/master/nginx.conf - the place where I can set those redirects or you mean to just preserve the original `_index.md` of the affected page and set the redirect there?

**Comment by vincent99** [comment]
Nginx is easier for bulk pattern matching/folder renames, [aliases](https://gohugo.io/content-management/urls/#aliases) for individual pages.

**Comment by macedogm** [comment]
@vincent99 thanks again for the tip. I added aliases for the removed pages. No links will be harmed now 😂

File: content/rancher/v2.6/en/cis-scans/_index.md [hunk 1]
```diff
@@ -1,289 +1,11 @@
...
-A report is generated with the scan results every time the scan runs. To see the latest results, click the name of the scan that appears.
-
-### Creating a Custom Benchmark Version for Running a Cluster Scan 
- 
-There could be some Kubernetes cluster setups that require custom configurations of the Benchmark tests. For example, the path to the Kubernetes config files or certs might be different than the standard location where the upstream CIS Benchmarks look for them.
-
-It is now possible to create a custom Benchmark Version for running a cluster scan using the `rancher-cis-benchmark` application.
-
-For details, see [this page.](./custom-benchmark)
+The CIS scans documentation is now part of the [Security > CIS Scans]({{<baseurl>}}/rancher/v2.6/en/security/cis-scans) section.
```
Comment (cbron) [inline content/rancher/v2.6/en/cis-scans/_index.md:11]: We do recommend hardening as a best practice, not just as a security step. Other items like logging are at top level menu level, so its more of a design question for docs team.

Comment (macedogm) [reply content/rancher/v2.6/en/cis-scans/_index.md:11]: Looking from this point of view, indeed it makes sense to keep the CIS docs as a top level section. I'll revert the removal of that section.

**Pull request merged**